lib/firmata.js

```diff
@@ -209,20 +211,28 @@ SYSEX_RESPONSE[CAPABILITY_RESPONSE] = function(board) {
 
   // Only create pins if none have been previously created on the instance.
   if (!board.pins.length) {
+    var pin = {};
     for (var i = 2, n = 0; i < board.currentBuffer.length - 1; i++) {
       if (board.currentBuffer[i] === 127) {
-        board.pins.push({
-          supportedModes: supportedModes(capability),
-          mode: undefined,
-          value: 0,
-          report: 1,
-        });
+        pin.supportedModes = supportedModes(capability);
```

Comment:
I'm good with the pin resolution changes, however I've been rethinking how to represent the Serial pins. I need to think more about how that information would be used (since you don't really need to think about pins when using a UART - you just need to know if it's Serial1, Serial2, Serial3, etc). One option is to revert the Serial pin changes for now and revisit once I have a better idea how to use them.